hydra-node/src/Hydra/HeadLogic.hs

```diff
@@ -1143,31 +1281,16 @@ newSn Environment{party} parameters CoordinatedHeadState{confirmedSnapshot, seen
 
 -- | Emit a snapshot if we are the next snapshot leader. 'Outcome' modifying
 -- signature so it can be chained with other 'update' functions.
-emitSnapshot :: IsTx tx => Environment -> Outcome tx -> Outcome tx
-emitSnapshot env outcome =
-  case outcome of
-    NewState (Open OpenState{parameters, coordinatedHeadState, chainState, headId, currentSlot}) ->
-      case newSn env parameters coordinatedHeadState of
-        ShouldSnapshot sn txs -> do
-          let CoordinatedHeadState{seenSnapshot} = coordinatedHeadState
-          NewState
-            ( Open
-                OpenState
-                  { parameters
-                  , coordinatedHeadState =
-                      coordinatedHeadState
-                        { seenSnapshot =
-                            RequestedSnapshot
-                              { lastSeen = seenSnapshotNumber seenSnapshot
-                              , requested = sn
-                              }
-                        }
-                  , chainState
-                  , headId
-                  , currentSlot
-                  }
-            )
-            `Combined` Effects [NetworkEffect (ReqSn sn (txId <$> txs))]
-        _ -> outcome
-    Combined l r -> Combined (emitSnapshot env l) (emitSnapshot env r)
-    _ -> outcome
+emitSnapshot ::
+  IsTx tx =>
+  Environment ->
+  HeadParameters ->
+  CoordinatedHeadState tx ->
+  SeenSnapshot tx ->
+  Outcome tx
+emitSnapshot env params chs seenSnapshot =
+  case newSn env params chs of
```

Comment:
We must challenge this function `newSn` because it's using an already update state and this has some nasty implications.

Also note that it means we would need to update the spec maybe.